### PR TITLE
Improve descriptions in localrules.yaml

### PR DESCRIPTION
**Pull request by anthonyeleven**
docs: Fix multiple spelling errors, improve wording

Upstream changes are in https://github.com/ceph/ceph/pull/47284

Signed-off-by: Anthony D'Atri <anthonyeleven@users.noreply.github.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by subhamkrai — COMMENTED** [review]
This file is copied from https://github.com/ceph/ceph/blob/master/monitoring/ceph-mixin/prometheus_alerts.yml and also simialr changes are requiredd from `deploy/examples/monitoring/localrules.yaml`

**Review by travisn — CHANGES_REQUESTED** [review]
The changes look great, we will just need to update them in the ceph repo as mentioned by @subhamkrai so we don't diverge if possible. Would you mind opening a PR there as well?
The CI will be resolved by rebasing since #10630 was merged.

**Comment by anthonyeleven** [comment]
Tests are still timing out :(

**Comment by travisn** [comment]
> Tests are still timing out :(

Looks like a flaky run, I triggered them again now just in case.

**Comment by anthonyeleven** [comment]
> This file is copied from https://github.com/ceph/ceph/blob/master/monitoring/ceph-mixin/prometheus_alerts.yml and also similar changes are required from `deploy/examples/monitoring/localrules.yaml`

I'll do an upstream `ceph/ceph` PR, and will iterate through  `deploy/examples/monitoring/localrules.yaml` .  In the meantime, can you help me understand why `deploy/examples/monitoring/localrules.yaml` isn't redundant given `deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml` ?

**Comment by travisn** [comment]
> > This file is copied from https://github.com/ceph/ceph/blob/master/monitoring/ceph-mixin/prometheus_alerts.yml and also similar changes are required from `deploy/examples/monitoring/localrules.yaml`
> 
> I'll do an upstream `ceph/ceph` PR, and will iterate through `deploy/examples/monitoring/localrules.yaml` . In the meantime, can you help me understand why `deploy/examples/monitoring/localrules.yaml` isn't redundant given `deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml` ?

The localrules.yaml in the two different places is for the helm and non-helm implementation that has slightly different implementation for pulling the rules in. For example, the helm chart has a setting that switches between external or local rules. It would certainly be better if we avoided this redundancy, but for now it exists. You should be able to update one location, then copy to the other, with the only difference being indentation.

> Congruence with ceph/ceph is somewhat problematic, eg. in `CephMonDiskspaceLow` a default path is listed that is inaccurate for Rook deployments.

Yeah we can't be fully in sync either since cephadm rules don't apply, but hoping we can stay in sync as much as possible.

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml [hunk 1]
```diff
@@ -698,37 +705,61 @@ groups:
...
-          summary: Freespace in a pool is too low for recovery/rebalance
+          summary: Free space in a pool is too low for recovery/rebalance
           description: >
-            A pool is approaching it's near full threshold, which will
+            A pool is approaching the near full threshold, which will
             prevent rebalance operations from completing. You should
-            consider adding more capacity to the pool.
+            consider adding more capacity.
 
+      - alert: CephPoolBackfillFull
```
Comment (travisn) [inline deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml:752]: This alert looks duplicated from the one before it

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml [hunk 2]
```diff
@@ -698,37 +705,61 @@ groups:
...
         annotations:
-          summary: MTU settings across Ceph hosts are inconsistent
+          summary: MTU settings across hosts are inconsistent
           description: >
             Node {{ $labels.instance }} has a different MTU size ({{ $value }})
             than the median value on device {{ $labels.device }}.
 
   - name: pools
     rules:
+      - alert: CephPoolGrowthWarning
```
Comment (travisn) [inline deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml:727]: Did you define this alert, or it's picked up from the ceph repo?

**Review by travisn — CHANGES_REQUESTED** [review]
I see the ceph PR opened https://github.com/ceph/ceph/pull/47284, thanks for mirroring it there. Just a couple final questions...

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml (hunk 2 above)
Comment (anthonyeleven) [reply deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml:728]: The latter, I noticed that it was missing from Rook.

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml (hunk 1 above)
Comment (anthonyeleven) [reply deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml]: Editing flern I'm sure, pushed a fix.

File: deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml (hunk 2 above)
Comment (travisn) [reply deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml:728]: It's definitely been a few months since we picked it up, sounds good to add it

**Review by travisn — APPROVED** [review]

**Pull request merged**